barchartrealtime.js

```diff
@@ -0,0 +1,135 @@
+// select the svg container first
+const svg = d3.select('.canvas')
+  .append('svg')
+    .attr('width', 600)
+    .attr('height', 600);
+//create the Y label
+svg.append("text")
+    .attr("text-anchor", "end")
+    .attr("x", 50)
+    .attr("y", 250)
+    .text("Sms(s)") 
+
+// create margins & dimensions
+const margin = {top: 20, right: 20, bottom: 100, left: 100};
+const graphWidth = 600 - margin.left - margin.right;
+const graphHeight = 600 - margin.top - margin.bottom;
+
+const graph = svg.append('g')
+  .attr('width', graphWidth)
+  .attr('height', graphHeight)
+  .attr('transform', `translate(${margin.left}, ${margin.top})`);
+
+// create axes groups
+const xAxisGroup = graph.append('g')
+  .attr('transform', `translate(0, ${graphHeight})`)
+
+xAxisGroup.selectAll('text')
+  .attr('fill', d => d.fill)
+  .attr('transform', 'rotate(-40)')
+  .attr('text-anchor', 'end');
+
+const yAxisGroup = graph.append('g');
+
+const y = d3.scaleLinear()
+    .range([graphHeight, 0]);
+
+const x = d3.scaleBand()
+  .range([0, graphWidth])
+  .paddingInner(0.2)
+  .paddingOuter(0.2);
+
+// create & call axes
+const xAxis = d3.axisBottom(x);
+const yAxis = d3.axisLeft(y)
+  .ticks(5)
+  .tickFormat(d => d + ' sms');
+
+  const t =d3.transition().duration(1500)
+
+// the update function
+const update = (data) => {
+
+  // 1. join the data to rects
+  const rects = graph.selectAll('rect')
+    .data(data);
+
+  console.log(rects);
+
+  //2. remove unwanted rects
+  rects.exit().remove();
+
+  //3. update the domains
+  y.domain([0, d3.max(data, d => d.sms)]);
+  x.domain(data.map(item => item.county));
+
+  //4. add attrs to rects already in the DOM
+  rects.attr('width', x.bandwidth)
+    .attr('fill', d => d.fill)
+    .attr('x', d => x(d.county))
+
+    /*
+    .transition().duration(500)
+        .attr('height',d => graphHeight - y(d.sms))
+        .attr('y', d => y(d.sms)); */
+
+  //4.1 append the enter selection to the DOM
```

Comment:
I think each of these would probably be better separated into different functions?